packages/core/src/components/essentials/legend.ts

```diff
@@ -111,28 +83,73 @@ export class Legend extends Component {
 			.classed('active', function (d, i) {
 				return d.status === Configuration.legend.items.status.ACTIVE;
 			});
+
 		const addedLegendItemsText = addedLegendItems
 			.append('text')
 			.merge(legendItems.select('text'));
 
-		// truncate the legend label if it's too long
-		if (truncationType !== TruncationTypes.NONE) {
-			addedLegendItemsText.html(function (d) {
-				if (d.name.length > truncationThreshold) {
-					return Tools.truncateLabel(
-						d.name,
-						truncationType,
-						truncationNumCharacter
-					);
-				} else {
-					return d.name;
-				}
-			});
-		} else {
-			addedLegendItemsText.html((d) => d.name);
+		this.truncateLegendText(addedLegendItemsText);
+
+		// Keep track of line numbers and positions
+		const itemConfig = {
+			startingPoint: 0,
+			itemIndexInLine: 0,
+			lineNumber: 0,
+			lastLegendItemTextWidth: 0,
+		};
+
+		this.breakItemsIntoLines(addedLegendItems, 'legend-item', itemConfig);
+
+		const additionalItemsOption = Tools.getProperty(
+			options,
+			'legend',
+			'additionalItems'
+		);
+
+		// Update radius label text with option
+		const radiusLabel = Tools.getProperty(
+			this.getOptions(),
+			'bubble',
+			'radiusLabel'
+		);
+		if (radiusLabel) {
+			additionalItemsOption[0].name = radiusLabel;
 		}
```

Comment:
this wont make sense because other charts might use the radius label (ex circle pack)